### PR TITLE
doc:  fixup -coinstatsindex help, update bitcoin.conf and files.md

### PR DESCRIPTION
**Pull request by jonatack**

### CODE REVIEW

**Comment by hebasto** [comment]
It'd would be nice to update the [`doc/files.md`](https://github.com/bitcoin/bitcoin/blob/master/doc/files.md) as well:

https://github.com/bitcoin/bitcoin/blob/2b45cf0bcdb3d2c1de46899e30885c953b57b475/src/index/coinstatsindex.cpp#L100

**Comment by jonatack** [comment]
Good idea @hebasto. Thanks, done.

**Review by benthecarman — COMMENTED** [review]
CR-ACK 15796fb87a2f03edc94668eeedaef5249c319f44

File: src/init.cpp [hunk 1]
```diff
@@ -384,7 +384,7 @@ void SetupServerArgs(NodeContext& node)
 #endif
     argsman.AddArg("-blockreconstructionextratxn=<n>", strprintf("Extra transactions to keep in memory for compact block reconstructions (default: %u)", DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg("-blocksonly", strprintf("Whether to reject transactions from network peers. Automatic broadcast and rebroadcast of any transactions from inbound peers is disabled, unless the peer has the 'forcerelay' permission. RPC transactions are not affected. (default: %u)", DEFAULT_BLOCKSONLY), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
-    argsman.AddArg("-coinstatsindex", strprintf("Maintain coinstats index used by the gettxoutset RPC (default: %u)", DEFAULT_COINSTATSINDEX), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```
Comment (MarcoFalke) [inline src/init.cpp:387]: test/functional/rpc_blockchain.py:        # muhash should not be included in gettxoutset unless requested.

File: share/examples/bitcoin.conf [hunk 2]
```diff
@@ -142,8 +142,11 @@
 # both prior transactions and several dozen future transactions.
 #keypool=100
 
+# Maintain coinstats index used by the gettxoutsetinfo RPC (default: 0).
+#coinstatsindex=1
+
 # Enable pruning to reduce storage requirements by deleting old blocks.
-# This mode is incompatible with -txindex and -rescan.
+# This mode is incompatible with -txindex, -coinstatsindex and -rescan.
```
Comment (MarcoFalke) [inline share/examples/bitcoin.conf:149]: This is wrong. Indexes are compatible with pruning, if done right.

File: src/init.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/init.cpp:387]: (same mistake in the test, that can be fixed as well)

File: share/examples/bitcoin.conf (hunk 2 above)
Comment (jonatack) [reply share/examples/bitcoin.conf:149]: If I understand correctly, @fjahr is working on that but for the moment it isn't the case yet

src/init.cpp#L888
```cpp
    argsman.AddArg("-prune=<n>", strprintf("Reduce storage requirements by enabling pruning (deleting) of old blocks. This allows the pruneblockchain RPC to be called to delete specific blocks, and enables automatic pruning of old blocks if a target size in MiB is provided.
            This mode is incompatible with -txindex, -coinstatsindex and -rescan. "
            "Warning: Reverting this setting requires re-downloading the entire blockchain. "
            "(default: 0 = disable pruning blocks, 1 = allow manual pruning via RPC, >=%u = automatically prune block files to stay under the specified target size in MiB)", MIN_DISK_SPACE_FOR_BLOCK_FILES / 
```


src/init.cpp#L888
```cpp
    // If using block pruning, then disallow txindex and coinstatsindex
    if (args.GetArg("-prune", 0)) {
        if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX))
            return InitError(_("Prune mode is incompatible with -txindex."));
        if (args.GetBoolArg("-coinstatsindex", DEFAULT_COINSTATSINDEX))
            return InitError(_("Prune mode is incompatible with -coinstatsindex."));
    }
```

Comment (MarcoFalke) [reply share/examples/bitcoin.conf:149]: #15946

Comment (jonatack) [reply share/examples/bitcoin.conf:149]: In principle it seems it should be doable: https://github.com/bitcoin/bitcoin/pull/19521/#discussion_r603607663 (not sure this GitHub link is working)

File: src/init.cpp (hunk 1 above)
Comment (jonatack) [reply src/init.cpp:387]: updating

**Comment by Sjors** [comment]
utACK 54133c59b80ccac85eaebb0668cd2f0fe360b323

**Comment by MarcoFalke** [comment]
cr ACK 54133c59b80ccac85eaebb0668cd2f0fe360b323

**Comment by clarkmoody** [comment]
utACK 54133c5

**Pull request merged**